### DISCUSSION TITLE
the doubts about the different annotations of BCR in CSR using different versions of MiXCR

### DISCUSSION THREAD

**Discussion by bigcat1001** [milaboratory/mixcr#1865] [Q&A] [answered]
Hello, I am sequencing the CDR3 of BCR using PE150, and my main focus is on CSR (Class Switch Recombination). The definition of a CSR clone is that the V gene and J gene are the same, the CDR3 nt sequence is identical, but the C region is different. However, I encountered a problem when analyzing the data with MixCR: When I used version 4.1.2 for analysis, the results included clones with CSR. But when I used version 4.6.0, there were no CSR clones. What could be the reason for this? Here are the program settings I used for the analysis with 4.1.2:
```
mixcr.jar analyze  takara-human-bcr-cdr3 --threads 40 \
	--species hsa \
	--rna \
	--rigid-left-alignment-boundary \
	--floating-right-alignment-boundary C \
	--tag-pattern "^N{7}(R1:*)\^(UMI:NNNNTtNNNNTNNNN)(R2:*)" \
	$i1 \
	$i2 \
	${i%/}_14B1_4.12
```

Here are the program settings I used for the analysis with 4.6.0:
```
mixcr.jar analyze  generic-amplicon-with-umi --threads 40 \
	--species hsa \
	--rna \
	--rigid-left-alignment-boundary \
	--floating-right-alignment-boundary C \
	--tag-pattern "^N{7}(R1:*)\^(UMI:NNNNTtNNNNTNNNN)(R2:*)" \
	$i1 \
	$i2 \
	${i%/}_14B1_4.6

```
Here are the clones with CSR from the analysis results using version 4.1.2:
[CSR clonetypes.csv](https://github.com/user-attachments/files/18081445/CSR.clonetypes.csv)

Thanks for your help!

**Comment by mizraelson** [comment, marked as answer]
Hi, the Takara preset includes the following parameter: `--split-clones-by C`.  If you add it to your command it should work.

**Comment by bigcat1001** [comment]
Thanks for your help.It works.
